app/models/manageiq/providers/vmware/infra_manager/provision/configuration/network.rb

```diff
@@ -96,6 +98,38 @@ def build_config_spec_dvs(network, vnicDev, vmcs)
     end
   end
 
+  def build_config_spec_opaque(network, vnicDev, vmcs)
+    hosts = dest_cluster.try(:hosts) || dest_host
+    lan = Lan.find_by(:name => network[:network], :switch_id => HostSwitch.where(:host_id => hosts).pluck(:switch_id))
```

Comment:
Is this query valid for opaque case? @agrare

Reply:
Yeah looks right